packages/block-editor/src/hooks/margin.js

```diff
@@ -52,7 +52,7 @@ export function MarginEdit( props ) {
 	const units = useCustomUnits();
 	const sides = useCustomSides( blockName, 'margin' );
 
-	if ( ! hasMarginSupport( blockName ) ) {
+	if ( useIsMarginDisabled( { name: blockName } ) ) {
```

Comment:
You could just pass through the `props` here. Maybe that is a little easier to read?

```suggestion
	if ( useIsMarginDisabled( props ) ) {
```